File: C/happy_number.c
bool isHappy(int n){
    if (n == 1) return 1;
    if (n == 4) return 0;
        
    int sum = 0;
        
    while (n != 0) {
        sum += (int)(pow(n % 10, 2));
        n /= 10;
    }
        
    return isHappy(sum);
}

File: Python3/happy_number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        loop = []
        
        while True:
            total = sum([pow(int(x), 2) for x in str(n)])         
            if total == 1: return True
            if total in loop: return False
            loop.append(total)
            n = total
